source/RSRCFilter.cpp

```diff
@@ -0,0 +1,43 @@
+#include "RSRCFilter.h"
+
+#include <MimeType.h>
+#include <string.h>
+
+static const char* supported_types[] = {
+	"application/x-vnd.be-elfexecutable",
+	"application/octet-stream",
+	NULL
+};
+
+bool
+RSRCFilter::Filter(const entry_ref* ref, BNode* node, struct stat_beos* st,
+	const char* filetype)
+{
+	BEntry entry(ref, true);
+	if (entry.InitCheck() != B_OK)
+		return false;
+	if (entry.IsDirectory())
+		return true;
+
+	entry_ref entryRef;
+	if (entry.GetRef(&entryRef) != B_OK)
+		return false;
+
+	BMimeType mimeType;
+	if (BMimeType::GuessMimeType(&entryRef, &mimeType) != B_OK)
```

Comment:
Only do the guessing if filetype is empty or "application/octet-stream". If you do have to guess, use the guessed mime type as "filetype".
Maybe add a comment that "All files of non-BFS drive report as octet-stream or emtpy".